module/platform/darwin/SensorFilter/src/SensorFilter.cpp

```diff
@@ -491,43 +514,37 @@ namespace module::platform::darwin {
                                              config.motionFilter.noise.measurement.flatFootOrientation,
                                              MeasurementType::FLAT_FOOT_ORIENTATION());
                     }
+                    // Else the foot is off the ground, so we make sure that for the next time step, we know that
+                    // this time step, the foot was off the ground
                     else if (!foot_down) {
                         previous_foot_down[side] = false;
                     }
 
+                    // Note that the Hwf is set, even if the foot is not down. This means that moving feet in the air
+                    // will have an Hwf associated with them which is the transform from when that foot last hit
+                    // the ground
                     sensors->feet[side].Hwf = footlanding_Hwf[side].matrix();
                 }
 
                 // Calculate our time offset from the last read
-                double deltaT = std::max(
+                // too hard to read
+                const double deltaT = std::max(
                     (input.timestamp - (previousSensors ? previousSensors->timestamp : input.timestamp)).count()
                         / double(NUClear::clock::period::den),
                     0.0);
```

Comment:
```suggestion
                using namespace std::chrono;
                const double deltaT = std::max(
                    duration_cast<duration<double>>(input.timestamp - (previousSensors ? previousSensors->timestamp : input.timestamp)).count(), 0.0);
```

Reply:
You've omitted the `den` term - was that deliberate?

Reply:
I omitted the `num` term, and that was technically lazyness. because nano/pico/milli etc all have a 1 as the num term. Technically shouldn't do that

Reply:
In the previous calculation it also had a `den` term - I don't think those are 1 :p

Reply:
I have gone with multiplying it by a constexpr double which is `nuclear::..::num / nuclear::...::den`. lmk if that's incorrect